other/travis/hstox-install

```diff
@@ -8,16 +8,12 @@ cabal update
 # install.
 cabal install happy
 
-# Fetch hstox.
-git clone --recursive --depth=1 https://github.com/TokTok/hstox ../hstox
+# Fetch hstox and msgpack-c.
+git clone --recursive --depth=1 https://github.com/TokTok/hs-toxcore ../hs-toxcore
+git clone --depth=1 https://github.com/msgpack/msgpack-c testing/hstox/msgpack-c
```

Comment:
The hstox build is the shortest of the 5, with 2 minutes 38 seconds on the latest build, vs. 4-5 minutes for the others. It wouldn't help and just add complexity at this point. Perhaps if hstox becomes very large and compiling it takes significant time (right now it's about 10 seconds), we can add caching.

Oh, and msgpack-c is not built. It's compiled directly as part of building the SUT. Installing it and caching its install would take more time than just compiling the 5 extra C files every time.